core/token_test.go

```diff
@@ -290,12 +291,31 @@ func NewTestTokenValidationContext(t *testing.T) (*httptest.Server, *util.HttpCo
 func TestCanValidateToken(t *testing.T) {
 	srv, ctx := NewTestTokenValidationContext(t)
 	defer srv.Close()
-	token := generateToken(t, time.Now(), time.Now().AddDate(0, 0, 1), srv.URL+"/SAAS/auth")
+	token := generateToken(t, time.Now(), time.Now().AddDate(0, 0, 1), srv.URL+"/SAAS/auth",
+		"")
 	new(TokenService).ValidateIDToken(ctx, token)
 	AssertOnlyInfoContains(t, ctx, "ID token is valid")
 	AssertOnlyInfoContains(t, ctx, "iss: "+srv.URL+"/SAAS/auth")
 }
 
+func TestCanRetrieveUsername(t *testing.T) {
+	srv, ctx := NewTestTokenValidationContext(t)
+	defer srv.Close()
+	token := generateToken(t, time.Now(), time.Now().AddDate(0, 0, 1), srv.URL+"/SAAS/auth",
+		"foobar@MY")
+	var username (string) = new(TokenService).ExtractUserIDFromIDToken(ctx, token)
+	assert.Equal(t, username, "foobar@MY")
+}
+
+func TestBadTokenGetUsername(t *testing.T) {
```

Comment:
Why is it bad? Seems like the assertEqual just checks the exact same thing as the previous test (that whatever is in `sub` is extracted as the username)